src/main/java/entities/Task.java

```diff
@@ -1,221 +1,221 @@
-        this.tags.add(tag);
-    }
-
-    /**
-     * Removes a tag from this Task
-     *
-     * @param tag The new tag to add
-     */
-    public void removeTag(Tag tag) {
-        this.tags.remove(tag);
-    }
-
-    /**
-     * Gets a list of tags associated with this Task
-     *
-     * @return tags A list of tags associated with this Task
-     */
-    public ArrayList<Tag> getTags() {
-        return this.tags;
-    }
-
-    /**
-     * Gets the User of this Task
-     */
-    public User getUser() {
-        return this.user;
-    }
-
-
-    /**
-     * Sets this task as completed
-     */
-    public void setCompleted() {
-        this.completed = true;
-    }
-
-    /**
-     * Sets this task as incompleted
-     */
-    public void setIncompleted() {
-        this.completed = false;
-    }
-
-    /**
-     * Gets the completion status of this task
-     *
-     * @return If the task is completed
-     */
-    public Boolean getCompleted() {
-        return this.completed;
-    }
-
-    //    Event not yet implemented
-//    /**
-//     * Adds an event to this Task
-//     * @param tag The new tag to add
-//     */
-//    public void addEvent(Event event) {
-//        this.events.add(tag);
-//    }
-//
-//    /**
-//     * Removes an event from this Task
-//     * @param tag The new tag to add
-//     */
-//    public void removeEvent(Event event) {
-//        this.events.remove(tag);
-//    }
-//
-//    /**
-//     * Gets a list of events associated with this Task
-//     * @return events A list of events associated with this Task
-//     */
-//    public ArrayList<Tag> getEvents() {
-//        return this.events;
-//    }
-//
-
-    //    Collaborator not yet implemented
-//    /**
-//     * Adds a collaborator to this Task
-//     * @param collaborator The collaborator to add to this task
-//     */
-//    public void addCollaborator(User collaborator) {
-//        this.collaborators.add(collaborator);
-//    }
-//
-//    /**
-//     * Removes a collaborator to this Task
-//     * @param collaborator The collaborator to remove from this task
-//     */
-//    public void removeCollaborator(User collaborator) {
-//        this.collaborators.remove(collaborator);
-//    }
-//
-//    /**
-//     * Gets the collaborators of this Task
-//     * @return
-//     */
-//    public ArrayList<User> addCollaborator(User collaborator) {
-//        return this.collaborators;
-//    }
-
-}
-
-
+
+package entities;
+
+import java.util.ArrayList;
+
+/**
+ * An entity class representing a Task
+ */
+public class Task {
+
+    /**
+     * The User that this Task belongs to
+     */
+    final User user;
+    /**
+     * The name of the Task
+     */
+    String name;
+
+    /**
+     * The description of the Task
+     */
+    String description;
+
+    /**
+     * A boolean indicating if the Task is completed
+     */
+    Boolean completed;
+
+    /**
+     * A list of the associated tags of this Task
+     */
+    ArrayList<Tag> tags;
```

Comment:
I know that this is Leon's code which I have asked him to change, so I will not comment on possible code smells regarding his code.